### PR TITLE
[158][164][198] Implement casting transaction list page

### PR DESCRIPTION
**Pull request by miello**
## What did you do

- Refactor layout in admin transaction page
- Implement casting transaction list page
- Extend mockApiClient functionality to support put, post

## Demo

- Login as admin (any account is okay) first
- Go to https://dev.modela.miello.dev/transaction/casting

1. Try accept or reject transaction (which should be gone after submitted)
2. Try SideMenu that navigates correctly
3. Try to see details of casting transaction

image

### CODE REVIEW

**Comment by PKhing** [comment]
> Go to https://dev.modela.ac.th/transaction/casting 

.ac.th 5555555555555555555

**Comment by PKhing** [comment]
image

File: apps/web/src/modules/transaction/pages/casting/components/ConfirmModalBody/constants.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export const MODAL_OPTION = {
+  accept: {
+    title: 'ทำเครื่องหมายว่าเสร็จสิ้นแล้ว',
+    finalState: 'จัดการเสร็จสิ้น',
+  },
+  reject: {
+    title: 'ปฏิเสธคำขอธุรกรรม',
+    finalState: 'ถูกปฏิเสธ',
+  },
+} as const
```
Comment (PKhing) [inline apps/web/src/modules/transaction/pages/casting/components/ConfirmModalBody/constants.ts:10]: why `as const` ?

Comment (miello) [reply apps/web/src/modules/transaction/pages/casting/components/ConfirmModalBody/constants.ts:10]: At first, I try to make the menubar type-strict which I have given up now 😂

**Comment by miello** [comment]
> image อันนี้ยังไม่หายเลย 😂 ลด width card ?

แก้ไปแล้ว แก้ใน Chrome Devtools (เรื่องจริง)

**Review by PKhing — APPROVED** [review]

**Pull request merged**